File: dotnet/audit-service/Models/Audit.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;

namespace audit_service.Models
{
    public class Audit : Identifiable<int>
    {
        [Attr] public string Tenant { get; set; }

        [Attr] public string Action { get; set; }

        [HasOne] public string Subject { get; set; }

        [HasOne] public string Object { get; set; }
    }
}

File: java/products-service/src/main/java/com/octopus/octopub/models/Audit.java
package com.octopus.octopub.models;

import com.github.jasminb.jsonapi.annotations.Id;
import com.github.jasminb.jsonapi.annotations.Type;
import lombok.Data;

@Type("audits")
@Data
public class Audit {

  @Id
  private String id;
  private String subject;
  private String action;
  private String object;

  public Audit(final String subject, final String action, final String object) {
    this.subject = subject;
    this.action = action;
    this.object = object;
  }
}
